src/Models/boundary_adjacent_mean.jl

```diff
@@ -0,0 +1,156 @@
+    └── max=0.980785, min=-0.980785, mean=1.11077e-16
+
+julia> bam(:east, cf)
+-7.806255641895632e-19
+
+julia> ff = Field{Face, Center, Center}(grid)
+17×16×16 Field{Face, Center, Center} on RectilinearGrid on CPU
+├── grid: 16×16×16 RectilinearGrid{Float64, Bounded, Periodic, Bounded} on CPU with 3×3×3 halo
+├── boundary conditions: FieldBoundaryConditions
+│   └── west: Nothing, east: Nothing, south: Periodic, north: Periodic, bottom: ZeroFlux, top: ZeroFlux, immersed: ZeroFlux
+└── data: 23×22×22 OffsetArray(::Array{Float64, 3}, -2:20, -2:19, -2:19) with eltype Float64 with indices -2:20×-2:19×-2:19
+    └── max=0.0, min=0.0, mean=0.0
+
+julia> set!(ff, (x, y, z) -> sin(2π * y / 4))
+17×16×16 Field{Face, Center, Center} on RectilinearGrid on CPU
+├── grid: 16×16×16 RectilinearGrid{Float64, Bounded, Periodic, Bounded} on CPU with 3×3×3 halo
+├── boundary conditions: FieldBoundaryConditions
+│   └── west: Nothing, east: Nothing, south: Periodic, north: Periodic, bottom: ZeroFlux, top: ZeroFlux, immersed: ZeroFlux
+└── data: 23×22×22 OffsetArray(::Array{Float64, 3}, -2:20, -2:19, -2:19) with eltype Float64 with indices -2:20×-2:19×-2:19
+    └── max=0.980785, min=-0.980785, mean=1.0284e-16
+
+julia> bam(:east, ff)
+-1.5612511283791264e-18
+
+julia> using Oceananigans.BoundaryConditions: PerturbationAdvectionOpenBoundaryCondition
+
+julia> obc = PerturbationAdvectionOpenBoundaryCondition(bam)
+┌ Warning: `PerturbationAdvection` open boundaries matching scheme is experimental and un-tested/validated
+└ @ Oceananigans.BoundaryConditions ~/Oceananigans.jl/src/BoundaryConditions/perturbation_advection_open_boundary_matching_scheme.jl:52
+OpenBoundaryCondition{Oceananigans.BoundaryConditions.PerturbationAdvection{Val{true}, Float64}}: BoundaryAdjacentMean: (-1.5612511283791264e-18)
+
+julia> u_bcs = FieldBoundaryConditions(east = obc)
+Oceananigans.FieldBoundaryConditions, with boundary conditions
+├── west: DefaultBoundaryCondition (FluxBoundaryCondition: Nothing)
+├── east: OpenBoundaryCondition{Oceananigans.BoundaryConditions.PerturbationAdvection{Val{true}, Float64}}: BoundaryAdjacentMean: (-1.5612511283791264e-18)
+├── south: DefaultBoundaryCondition (FluxBoundaryCondition: Nothing)
+├── north: DefaultBoundaryCondition (FluxBoundaryCondition: Nothing)
+├── bottom: DefaultBoundaryCondition (FluxBoundaryCondition: Nothing)
+├── top: DefaultBoundaryCondition (FluxBoundaryCondition: Nothing)
+└── immersed: DefaultBoundaryCondition (FluxBoundaryCondition: Nothing)
+
+```
+"""
+struct BoundaryAdjacentMean{NI, BV}
+  boundary_parallel_integral :: NI
+                       value :: BV
+
+   BoundaryAdjacentMean(grid, side; 
+                        boundary_parallel_integral::NI = boundary_reduced_field(Val(side), grid),
+                        value::BV = Ref(zero(grid))) where {NI, BV} = 
+        new{NI, BV}(boundary_parallel_integral, value)
+end
+
+@inline (bam::BoundaryAdjacentMean)(args...) = bam.value[]
```

Comment:
I just noticed this. I would offer that this code is surprising and hard to understand.